src/typechecker.ts

```
@@ -0,0 +1,1901 @@
+
+      // All numbers are castable to one another
+      return true;
+    };
+
+    case 'func': {
+      const fa = a;
+      const fb = b as FuncType;
+      if (fa.args.length != fb.args.length) return false;
+      if (fa.args.some((argA, idx) => types_are_equivalent(argA.type, fb.args[idx]!.type))) return false;
+      return types_are_equivalent(fa.returns, fb.returns);
+    };
```

Biome (2.1.2)

[error] This code will never be reached ...

... because this statement will return from the function beforehand


(lint/correctness/noUnreachable)